pkg/api/v1/project/integration.go

```diff
@@ -82,7 +82,7 @@ func (i Integration) ToAtlas(c client.Client, defaultNS string) (result *mongodb
 			return
 		}
 
-		*target, err = passwordField.ReadPassword(c, defaultNS)
```

Comment:
I think you need to change the `ReadPassword()` behavior. It was supposed to fallback to `defaultNS` when the namespace of the requested resource it's not available, however, it's actually uses the passed namespace

Reply:
Good point, thanks! Fixed

Reply:
There's still something strange. Look at `GetObject()`, it is supposed to already return the correct object key(falling back to parent namespace) but seems to not work correctly.

Reply:
Yeah, looks it works the other way around. Let me fix this